examples/threading/threading.c

```diff
@@ -10,10 +10,30 @@
 
 void* threadfunc(void* thread_param)
 {
-
+    int rc;
     // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
     // hint: use a cast like the one below to obtain thread arguments from your parameter
     //struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+    struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+
+    usleep((thread_func_args-> wait_to_obtain_ms)*1000);
+
+    rc = pthread_mutex_lock(thread_func_args->mutex);
+    if(rc ==-1)
+    {
+        ERROR_LOG("Failure: Locking mutex ");
```

Comment:
@Ganeshkm10 will `thread_complete_success` indicate failure in this code path and others where a failure occurred?